CMakeLists.txt leaks compiler flags to linking projects

### What happened?

The CMakeLists.txt of Ark leaks compile options to dependent projects, causing them to fail compiling if they don't fulfill the same standards as the Ark codebase.

This is directly caused by using `PUBLIC` rather than `PRIVATE` in `target_compile_options` for `ArkReactor` this should rather be `PRIVATE` to avoid the leakage.
Here is a list of offending `target_compile_options`:

- [CMakeLists.txt:67](https://github.com/ArkScript-lang/Ark/blob/3ffb7730918d1fa66337dfe982ed392da20e44f8/CMakeLists.txt#L67)
- [CMakeLists.txt:85](https://github.com/ArkScript-lang/Ark/blob/3ffb7730918d1fa66337dfe982ed392da20e44f8/CMakeLists.txt#L85)
- [CMakeLists.txt:95](https://github.com/ArkScript-lang/Ark/blob/3ffb7730918d1fa66337dfe982ed392da20e44f8/CMakeLists.txt#L95C9-L95C31)
- [CMakeLists.txt:106](https://github.com/ArkScript-lang/Ark/blob/3ffb7730918d1fa66337dfe982ed392da20e44f8/CMakeLists.txt#L106)

Here is a simple example application that links with Ark, but can't compile due to it:
```cpp
int foo(int val) {
  auto lambda = [] {
    int val = 4; // <- Error due to shadowing foo's parameter
    return val;
  };
  return lambda();
}

int main() {
  return foo(4);
}
```
```cmake
project(test CXX)

set(ARK_BUILD_EXE     OFF)
set(ARK_BUILD_MODULES OFF)
set(ARK_ENABLE_SYSTEM OFF)
set(ARK_UNITY_BUILD   OFF)
add_subdirectory(Ark)

add_executable(MRE)
target_sources(MRE PUBLIC main.cpp)

target_include_directories(MRE PRIVATE ArkScript/include)
target_link_libraries(MRE PUBLIC ArkReactor)
target_compile_options(MRE PUBLIC "-std=c++20")
```

The error is directly related to the `-Werror -Wshadow` settings that Ark leaks into all linking application:
```
/home/justin/Projekte/ark-mre/main.cpp: In lambda function:
/home/justin/Projekte/ark-mre/main.cpp:6:9: error: declaration of 'int val' shadows a parameter [-Werror=shadow]
    6 |     int val = 4;
      |         ^~~
/home/justin/Projekte/ark-mre/main.cpp:4:13: note: shadowed declaration is here
    4 | int foo(int val) {
      |         ~~~~^~~
/home/justin/Projekte/ark-mre/main.cpp: In function 'int foo(int)':
/home/justin/Projekte/ark-mre/main.cpp:4:13: error: unused parameter 'val' [-Werror=unused-parameter]
```

### What ArkScript version are you seeing the problem on?

dev

### Operating System

Linux

### Compiler used

GCC

### Compiler version used

gcc-Version 14.2.1 20250207 (GCC) 

### Relevant code & log output

```shell

```

### Documentation

- [x] I have read existing related documentation before submitting this report
- [x] I have searched for an existing issue before posting

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -347,6 +347,7 @@ jobs:
         run: |
           mkdir crashes
           cp output/default/crashes/id:* crashes/
+          (cd crashes && find . -type f -exec mv '{}' '{}'.ark \;)
           tar -czvf crashes.tar.gz crashes
 
       - name: Upload crashes

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,7 +64,7 @@ if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR APPLE)
     target_compile_definitions(ArkReactor PRIVATE ARK_USE_COMPUTED_GOTOS=1)
 
     target_compile_options(ArkReactor
-            PUBLIC
+            PRIVATE
             -Wall -Wextra -pedantic -Wstrict-aliasing
             -Wshadow
             -Wconversion
@@ -82,8 +82,8 @@ if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR APPLE)
             -fno-semantic-interposition
     )
     if (NOT ARK_BENCHMARKS)
-        target_compile_options(ArkReactor PUBLIC -Werror)
-    endif()
+        target_compile_options(ArkReactor PRIVATE -Werror)
+    endif ()
 
     # set fmtlib visibility to default
     target_compile_definitions(ArkReactor PUBLIC FMT_SHARED)
@@ -92,7 +92,7 @@ if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR APPLE)
         # The standard SSH libraries are depreciate on APPLE.
         # Thus they currently generate a warning that we have to ignore for now.
         # Though we should look to fix that in the future.
-        target_compile_options(ArkReactor PUBLIC -Wno-deprecated-declarations)
+        target_compile_options(ArkReactor PRIVATE -Wno-deprecated-declarations)
     endif ()
 
     # The following sub-modules link with -L/usr/local/lib
@@ -104,21 +104,21 @@ if (CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG OR APPLE)
     #   lib/modules/msgpack/CMakeFiles/msgpack.dir/__/submodules/msgpack-cpp/src/vrefbuffer.c.o
     #   lib/modules/msgpack/CMakeFiles/msgpack.dir/__/submodules/msgpack-cpp/src/zone.c.o
     target_compile_options(ArkReactor
-            PUBLIC
+            PRIVATE
             -Wno-unused-command-line-argument)
 
     # The nlohmann/json.hpp external project has unused typedefs in the code
     # to compensate for this error we remove the following warning.
     if (CMAKE_COMPILER_IS_CLANG)
-        target_compile_options(ArkReactor PUBLIC -Wno-unused-local-typedef)
+        target_compile_options(ArkReactor PRIVATE -Wno-unused-local-typedef)
     elseif (CMAKE_COMPILER_IS_GNUCXX)
-        target_compile_options(ArkReactor PUBLIC -Wno-unused-local-typedefs)
+        target_compile_options(ArkReactor PRIVATE -Wno-unused-local-typedefs)
     endif ()
 
 elseif (MSVC)
     target_compile_definitions(ArkReactor PRIVATE ARK_USE_COMPUTED_GOTOS=0)
     target_compile_options(ArkReactor
-            PUBLIC
+            PRIVATE
             /W4
             /MP4      # build multiple source files concurrently
             /EHa      # set exception model to standard C++ stack unwinding
@@ -126,7 +126,8 @@ elseif (MSVC)
             /wd4244   # disable warning about data loss (size_t -> char)
             /wd4505   # disable warning about unused static function was deleted
             /wd4068   # disable warnings about unknown pragmas.
-            /utf-8)
+    )
+    target_compile_options(ArkReactor PUBLIC /utf-8)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /STACK:8000000")  # set stack size to 8MB
 endif ()
 
@@ -252,7 +253,7 @@ if (ARK_BENCHMARKS)
         CPMAddPackage(
                 URI "gh:CodSpeedHQ/codspeed-cpp#5324618a1a14e0aa9a99dcfb467b9c2a57957e19@1.1.1"
                 SOURCE_SUBDIR google_benchmark)
-    else()
+    else ()
         CPMAddPackage("gh:google/benchmark@1.8.3")
     endif ()
 

--- a/src/arkscript/main.cpp
+++ b/src/arkscript/main.cpp
@@ -17,7 +17,7 @@
 #ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
 #    define ARK_ERROR_EXIT_CODE 0
 #else
-#    define ARK_ERROR_EXIT_CODE -1
+#    define ARK_ERROR_EXIT_CODE (-1)
 #endif
 
 int main(int argc, char** argv)

